ddterm/shell/extension.js

```diff
@@ -140,6 +142,39 @@ class ExtensionDBusInterface {
     }
 }
 
+class DesktopEntry {
+    constructor() {
+        this.file = Gio.File.new_for_path(GLib.build_filenamev(
+            [
+                GLib.get_user_data_dir(),
+                'applications',
+                `${APP_ID}.desktop`,
+            ]
+        ));
+    }
+
+    install() {
+        try {
+            Gio.File.new_for_path(GLib.build_filenamev(
+                [
+                    GLib.get_user_data_dir(),
```

Comment:
Oh, there's another issue... The extension can also be installed system-wide. In this case, it won't be stored in user's directory. Please find a way to get the actual source directory. For example, look at how D-Bus interface XML is loaded.

Reply:
Yep! In addition to that, the right place to put the desktop file also won't be in the user's `$HOME` but on `/usr/share/applications`. Fortunately, there's `GLib.get_system_data_dirs` for that.

Will work on this next!

Reply:
You shouldn't need `GLib.get_system_data_dirs` or `GLib.get_user_data_dir`. Again, look at how D-Bus interface XML is loaded, in the same `extension.js`.

Reply:
Thanks, I'll use the same method to get the `source_file` that is the extension's directory like so:

```js       Me.dir.get_child('ddterm').get_child('com.github.amezin.ddterm.Extension.xml');
```

However, the `target_file` also changes location when the extension is installed system-wide. It shouldn't be installed in the user's local data directory, but rather the system data directory. For that, I think I'll need `GLib.get_system_data_dirs`.

Reply:
`Me.dir` is the extension's source directory. If the extension is installed system-wide, it's `/usr/share/gnome-shell/extensions/ddterm@...` (or whatever gnome-shell's prefix is), when installed in the "usual" way, in user's home, it is `~/.local/share/gnome-shell/extensions/ddterm@...`. There's no need for any additional logic.

Reply:
Got it now! Thanks for taking the time to explain.